ts-sdk/whirlpool/src/createPool.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  getFeeTierAddress,
+  getInitializePoolV2Instruction,
+  getInitializeTickArrayInstruction,
+  getTickArrayAddress,
+  getTickArraySize,
+  getTokenBadgeAddress,
+  getWhirlpoolAddress,
+  getWhirlpoolSize,
+} from "@orca-so/whirlpools-client";
+import type {
+  Address,
+  GetMinimumBalanceForRentExemptionApi,
+  GetMultipleAccountsApi,
+  IInstruction,
+  LamportsUnsafeBeyond2Pow53Minus1,
+  Rpc,
+  TransactionPartialSigner,
+} from "@solana/web3.js";
+import { generateKeyPairSigner } from "@solana/web3.js";
+import {
+  DEFAULT_ADDRESS,
+  DEFAULT_FUNDER,
+  SPLASH_POOL_TICK_SPACING,
+  WHIRLPOOLS_CONFIG_ADDRESS,
+} from "./config";
+import invariant from "tiny-invariant";
+import {
+  getFullRangeTickIndexes,
+  getTickArrayStartTickIndex,
+  priceToSqrtPrice,
+  sqrtPriceToTickIndex,
+} from "@orca-so/whirlpools-core";
+import { fetchAllMint, getTokenSize } from "@solana-program/token";
+
+type CreatePoolInstructions = {
+  instructions: IInstruction[];
+  initializationCost: LamportsUnsafeBeyond2Pow53Minus1;
+  poolAddress: Address;
+};
+
+export function createSplashPoolInstructions(
+  rpc: Rpc<GetMultipleAccountsApi & GetMinimumBalanceForRentExemptionApi>,
+  tokenMintOne: Address,
+  tokenMintTwo: Address,
+  initialPrice: number = 1,
+  funder: TransactionPartialSigner = DEFAULT_FUNDER,
+): Promise<CreatePoolInstructions> {
+  return createConcentratedLiquidityPoolInstructions(
+    rpc,
+    tokenMintOne,
+    tokenMintTwo,
+    SPLASH_POOL_TICK_SPACING,
+    initialPrice,
+    funder,
+  );
+}
+
+export async function createConcentratedLiquidityPoolInstructions(
+  rpc: Rpc<GetMultipleAccountsApi & GetMinimumBalanceForRentExemptionApi>,
+  tokenMintOne: Address,
+  tokenMintTwo: Address,
+  tickSpacing: number,
+  initialPrice: number = 1,
+  funder: TransactionPartialSigner = DEFAULT_FUNDER,
+): Promise<CreatePoolInstructions> {
+  invariant(
+    funder.address !== DEFAULT_ADDRESS,
+    "Either supply a funder or set the default funder",
+  );
+  const [tokenMintA, tokenMintB] =
```

Comment:
Since this is the high-level sdk I wouldn't want any consumer to have to worry about internal quirks of the program (e.g. token order). IMO flipping them if they are not the right way around wouldn't cause that much of an issue. I think adding a callout in the jsdoc comment should be sufficient @calintje